render/filters.go

```diff
@@ -44,12 +44,13 @@ func ColorConnected(r Renderer) Renderer {
 				}
 			}
 
+			output := input.Copy()
 			for id := range connected {
-				input[id] = input[id].WithNode(report.MakeNodeWith(map[string]string{
+				output[id] = output[id].WithNode(report.MakeNodeWith(map[string]string{
 					IsConnected: "true",
 				}))
 			}
-			return input
+			return output
```

Comment:
ah, true, that does matter. But my point was more that doing `input.Copy()` first is copying all connected nodes twice.